plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts

```diff
@@ -65,8 +71,31 @@ export class AuthorizedEntitiesCatalog implements EntitiesCatalog {
     return this.entitiesCatalog.entities(request);
   }
 
-  removeEntityByUid(uid: string): Promise<void> {
-    // TODO: Implement permissioning
+  async removeEntityByUid(
+    uid: string,
+    authorizationToken?: string,
+  ): Promise<void> {
+    const { entities } = await this.entitiesCatalog.entities({
+      filter: basicEntityFilter({ 'metadata.uid': uid }),
+    });
+    if (entities.length === 0) {
+      throw new NotFoundError('No such entity.');
```

Comment:
Maybe preferable to return a NotAllowedError here to prevent users from using this API to find entities that exist but aren't visible to them?

Reply:
Good point, done!